nixos/modules/services/misc/gammu-smsd.nix

```diff
@@ -0,0 +1,261 @@
+          description = "Owner group of the device";
+          example = "dialout";
+        };
+
+        connection = mkOption {
+          type = types.string;
+          default = "at";
+          description = "Protocol which will be used to talk to the phone";
+        };
+        
+        synchronizeTime = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Whether to set time from computer to the phone during starting connection";
+        };
+
+        pin = mkOption {
+          type = types.nullOr types.string;
+          default = null;
+          description = "PIN code for the simcard";
+        };
+      };
+
+
+      log = {
+        file = mkOption {
+          type = types.string;
+          default = "syslog";
+          description = "Path to file where information about communication will be stored";
+        };
+
+        format = mkOption {
+          type = types.string;
+          default = "textalldate";
+          description = "Determines what will be logged to the LogFile";
+                                                                                                    
+          example = [ "nothing" "text" "textall" "textalldate" "errors" "errorsdate" "binary" ];
+        };
+      };
+   
+
+      extraConfig = {
+        gammu = mkOption {
+          type = types.lines;
+          default = "";
+          description = "Extra config lines to be added into [gammu] section";
+        };
+
+
+        smsd = mkOption {
+          type = types.lines;
+          default = "";
+          description = "Extra config lines to be added into [smsd] section";
+        };
+      };
+
+
+      backend = {
+        service = mkOption {
+          type = types.string;
+          default = "null";
+          description = "Service to use to store sms data.";
+          example = [ "null" "files" "sql" ];
+        };
+
+        files = {
+          inboxPath = mkOption {
+            type = types.path;
+            default = "/var/spool/sms/inbox/";
+            description = "Where the received SMSes are stored";
+          };
+          
+          outboxPath = mkOption {
+            type = types.path;
+            default = "/var/spool/sms/outbox/";
+            description = "Where SMSes to be sent should be placed";
+          };
+
+          sentSMSPath = mkOption {
+            type = types.path;
+            default = "/var/spool/sms/sent/";
+            description = "Where the transmitted SMSes are placed";
+          };
+
+          errorSMSPath = mkOption {
+            type = types.path;
+            default = "/var/spool/sms/error/";
+            description = "Where SMSes with error in transmission is placed";
+          };
+        };
+
+        sql = {
+          driver = mkOption {
+            type = types.string;
+            default = "";
+            description = "BD driver to use";
+            example = [ "native_mysql" "native_pgsql" "odbc" "dbi" ];
```

Comment:
The type says string, but the example says list of strings. Which one is it? Can more than one database driver be enabled at once? If not, maybe use "enum" type instead. (Btw, types.string is deprecated, use types.str instead.) Also, BD => DB typo :in description -)